### ISSUE TITLE
Metrics name inconsistencies between Micrometer and Prometheus

### ISSUE THREAD

**Issue by ghusta** [brettwooldridge/HikariCP#1714]
I found inconsistencies in the metrics names between Micrometer and Prometheus, by using either `PrometheusMetricsTrackerFactory` or `MicrometerMetricsTrackerFactory` when configuring `HikariConfig`.

For exemple, using PrometheusMetricsTrackerFactory, we get these metrics : 
- `hikaricp_active_connections`
- `hikaricp_idle_connections`

When using MicrometerMetricsTrackerFactory we get these ones : 
- `hikaricp.connections.active`
- `hikaricp.connections.idle`

Which will be translated to Prometheus naming conventions if using _io.micrometer:micrometer-registry-prometheus_ : 
- `hikaricp_connections_active`
- `hikaricp_connections_idle`

It can be cumbersome when reusing or adapting dashboards like in Grafana.

But I don't know if it can be changed easily without breaking backwards compatibility...

**Comment by brettwooldridge** [comment]
@ghusta Yeah, that's rather unfortunate. Unfortunately, those two features were largely contributions, with some changes by me, and then even further pull requests over time from other developers, so I think there wasn't much consideration given to naming consistency between the two.

I do agree it seems difficult to change at this point without breaking existing dashboards.  But I am open to suggestions.

Just off of the top of my head, I could conceive of something like a system property, eg. ``hikaricp.metricsNaming2.0``, that if defined as ``true`` (i.e., ``-Dhikaricp.metricsNaming2.0=true``) would change the metrics naming *in one direction or the other* to be consistent.  Possibly over time, with enough notice to users, this could be changed to be the default, and the *current behavior* could be preserved by explicitly setting the property to ``false``.

It is hard to say which naming is "correct", but subjectively my feeling is that ``hikaricp.connections.active`` is better than ``hikaricp.active.connections``, because in a *hierarchical key space* ``hikaricp.connections.*`` is consistent (both ``active`` and ``idle`` pertaining to ``connections``).  Obviously, in Prometheus, this would be "flattened" .

If this approach would be taken, then the system property would control the metric naming in the ``PrometheusMetricsTrackerFactory``, toggling between ``hikaricp_active_connections`` and ``hikaricp_connections_active``, for example.

If you think this is a good approach, could you submit a pull request?

**Comment by ghusta** [comment]
Thanks for your reply.
I quite agree with you.

Concerning the metrics naming, I had a look at this pages : 
- https://micrometer.io/docs/concepts#_naming_meters
- https://prometheus.io/docs/practices/naming/

But it almost give examples.
One another interesting example is [org.hibernate:hibernate-micrometer](https://github.com/hibernate/hibernate-orm/tree/master/hibernate-micrometer).
So I think the best naming is going from more general to more specific, as you said.

I found your suggestion relevant, using a kind a feature flag with a system property.

I'll see if I have some time to work on this, but I can't promise anything yet !
